pkg/storage/streams/store.go

```diff
@@ -37,3 +51,192 @@ type Store interface {
 		recursive bool, limit int, metaFlags uint32) (items []ListItem,
 		more bool, err error)
 }
+
+type streamStore struct {
+	segments    segments.Store
+	segmentSize int64
+}
+
+// NewStreams stuff
+func NewStreams(segments segments.Store, segmentSize int64) (Store, error) {
+	if segmentSize <= 0 {
+		return nil, errors.New("segment size must be larger than 0")
```

Comment:
I'm not sure what will happen in the future- should we create a commons.go and create error classes there - instead of doing adhoc error.New()? What are your thoughts here?

Reply:
i would prefer at least replacing this with `errs.New()`